scarb/tests/build.rs

```diff
@@ -1108,3 +1108,88 @@ fn add_statements_functions_debug_info() {
         "Expected statements_functions info to be a map"
     );
 }
+
+#[test]
+fn add_statements_functions_debug_info_to_tests() {
+    let t = TempDir::new().unwrap();
+    ProjectBuilder::start()
+        .name("hello")
+        .lib_cairo(indoc! {r##"
+            #[starknet::interface]
+            pub trait IHelloStarknet<TContractState> {
+                fn increase_balance(ref self: TContractState, amount: felt252);
+                fn get_balance(self: @TContractState) -> felt252;
+            }
+
+            #[starknet::contract]
+            mod HelloStarknet {
+                #[storage]
+                struct Storage {
+                    balance: felt252,
+                }
+
+                #[abi(embed_v0)]
+                impl HelloStarknetImpl of super::IHelloStarknet<ContractState> {
+                    fn increase_balance(ref self: ContractState, amount: felt252) {
+                        assert(amount != 0, 'Amount cannot be 0');
+                        self.balance.write(self.balance.read() + amount);
+                    }
+
+                    fn get_balance(self: @ContractState) -> felt252 {
+                        self.balance.read()
+                    }
+                }
+            }
+
+            fn foo(mut shape: Span<usize>) -> usize {
+                let mut result: usize = 1;
+
+                loop {
+                    match shape.pop_front() {
+                        Option::Some(item) => { result *= *item; },
+                        Option::None => { break; }
+                    };
+                };
+
+                result
+            }
+
+            fn main() -> usize {
+                foo(array![1, 2].span())
+            }
+        "##})
+        .manifest_extra(indoc! {r#"
+            [[target.starknet-contract]]
+
+            [cairo]
+            unstable-add-statements-functions-debug-info = true
+        "#})
+        .dep_starknet()
+        .build(&t);
+    Scarb::quick_snapbox()
+        .arg("build")
+        .arg("--test")
+        .current_dir(&t)
+        .assert()
+        .success();
+
+    let lib_sierra_string = t
+        .child("target/dev/hello_unittest.test.sierra.json")
+        .read_to_string();
+    let lib_sierra = serde_json::from_str::<VersionedProgram>(&lib_sierra_string).unwrap();
+
+    assert!(
+        lib_sierra
+            .into_v1()
+            .unwrap()
+            .debug_info
+            .expect("Expected debug info to exist")
+            .annotations
+            .get("github.com/software-mansion/cairo-profiler")
+            .expect("Expected cairo-profiler annotations to exist")
+            .get("statements_functions")
+            .expect("Expected statements_functions info to exist")
+            .is_object(),
```

Comment:
[nit] Better to try to deserialize
https://github.com/software-mansion/scarb/blob/d32b7638ed6fcbb707f29482930b59bcbe5464af/extensions/scarb-snforge-test-collector/tests/test.rs#L473

Reply:
Done in: #1407